File: boj/#2606.cpp
#include<iostream>
#include<queue>
#include<vector>

using namespace std;

bool visited[101] = { false, };

void bfs(vector<pair<int, int>>& net, int& ans) {
	
	queue<int> q;

	q.push(1);
	visited[1] = true;

	while (!q.empty()) {
		int cur = q.front();
		q.pop();

		for (auto n : net) {
			if (n.first == cur && visited[n.second] == false) {
				q.push(n.second);
				ans++;
				visited[n.second] = true;
			}
			else if (n.second == cur && visited[n.first] == false) {
				q.push(n.first);
				ans++;
				visited[n.first] = true;
			}
		}
	}

	return;
}

int main() {

	int numCom, numNet;
	int x, y;

	vector<pair<int, int>> net;

	cin >> numCom >> numNet;

	for (int i = 0; i < numNet; i++) {
		cin >> x >> y;
		net.push_back(make_pair(x, y));
	}

	int ans = 0;
	bfs(net, ans);

	cout << ans;
	return 0;
}

File: boj/#2606.py
from collections import deque

def bfs():
    
    ans = 0
    q = deque()

    visited[1] = True
    q.append(1)

    while len(q) != 0:
        cur = q.popleft()

        for edge in adj_list:
            if edge[0] == cur and visited[edge[1]] == False:
                visited[edge[1]] = True
                ans += 1
                q.append(edge[1])
            elif edge[1] == cur and visited[edge[0]] == False:
                visited[edge[0]] = True
                ans += 1
                q.append(edge[0])

    return ans
    

N = int(input())
M = int(input())

adj_list = []
visited = [False for _ in range(N + 1)]

for _ in range(M):
    adj_list.append(list(map(int, input().split())))

print(bfs())